### PR TITLE
feat:add button component

### PR DESCRIPTION
**Pull request by shubhamkumar648**
Add variant of button
1. primary button
2. Primary
button

3.Link
4.Icon Button

### CODE REVIEW

File: styles/button.css [hunk 1]
```diff
@@ -16,38 +16,184 @@
...
   }
   
   .btn-install:hover {
-    
     color: hsl(122, 39%, 49%);
+}
+
+
+/* button component */
+
```
Comment (rahulcs754) [inline styles/button.css:39]: don't leave space

File: styles/button.css [hunk 2]
```diff
@@ -16,38 +16,184 @@
...
+  background-color: var(--success-hover);
+}
+
+.btn__primary-text {
+  background-color: transparent;
+  border-style: hidden
+}
+
+.btn__primary-text:hover {
+
```
Comment (rahulcs754) [inline styles/button.css]: same as above

File: styles/button.css [hunk 3]
```diff
@@ -16,38 +16,184 @@
...
+  border: 1px solid var(--success-color);
+}
+
+.btn__sucess_outline:hover {
+  color: var(--lightwhite-color);
+  background-color: var(--success-hover);
+}
+
+.link_N_Icon_button a {
+
```
Comment (rahulcs754) [inline styles/button.css]: same as above

File: styles/button.css [hunk 4]
```diff
@@ -16,38 +16,184 @@
...
+.link_btn{
+  padding: 1rem 2rem;
+}
+.icons_btn {
+  display: flex;
+  gap: 8px;
+}
+
+.float__container{
+
```
Comment (rahulcs754) [inline styles/button.css]: same as above

File: styles/button.css [hunk 5]
```diff
@@ -16,38 +16,184 @@
...
+  width: 50px;
+  height: 50px;
+  font-size: 1.125rem;
+  border-radius: 50%;
+  background-color: var(--primary-color);
+  color: var(--text-color);
+  z-index: 2;
+  }
+  .float__btn:hover{
+
```
Comment (rahulcs754) [inline styles/button.css:183]: use prettier and don't leave any space

**Review by rahulcs754 — COMMENTED** [review]
Please check change and try to use prettier to clean your code view

File: component/button.html [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+          <button class="button">
+            <a href="#" class="link__nostyle">Installment</a>
+          </button>
+
+          <button class="button">
+            <a href="../pages/documentation.html" class="link__nostyle"
+              >Documnetation
+            </a>
+          </button>
```
Comment (shuntzhere) [inline component/button.html]: There is a typo here.

File: component/button.html [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+        <hr />
+
+        <div class="aside__content p-1">
+          <span>
+            <a class="active link__nostyle" href="../component/avatar.html"
+              >Avatar</a
+            ></span
+          >
+
```
Comment (shuntzhere) [inline component/button.html]: Format this code for better readability.

File: component/button.html [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+          >
+          <span>
+            <a class="active link__nostyle" href="../component/button.html"
+              >button</a
+            ></span
+          >
+
+          <span> <a class="active link__nostyle" href="#home">Home</a></span>
+          <span> <a class="active link__nostyle" href="#home">Home</a></span>
+        </div>
```
Comment (shuntzhere) [inline component/button.html:79]: href takes a link so you can just leave a "#". No need to use #home.

File: styles/button.css [hunk 9]
```diff
@@ -16,38 +16,184 @@
...
+}
+
+.float__container{
+
+  position: static;
+}
+
+.float__btn{
+  box-sizing: border-box;
+  display: flex;
```
Comment (shuntzhere) [inline styles/button.css]: You don't need to give box-sizing here.

**Review by shuntzhere — CHANGES_REQUESTED** [review]
Make sure to format the code using extensions like prettier that would remove the extra lines too, which will thus improve the readability.
Also suggested some minor changes.

File: styles/button.css (hunk 9 above)
Comment (shubhamkumar648) [reply styles/button.css]: thank you forgot to remove

File: component/button.html (hunk 7 above)
Comment (shubhamkumar648) [reply component/button.html]: it will reflect  in next commit

**Comment by shubhamkumar648** [comment]
> Make sure to format the code using extensions like prettier that would remove the extra lines too, which will thus improve the readability. Also suggested some minor changes.

 yeah i used i use prettier it gives these formet

**Review by rahulcs754 — APPROVED** [review]

**Review by shuntzhere — APPROVED** [review]
Looks good now.

**Pull request merged**